### PR TITLE
Angular upgrade 10

### PR DESCRIPTION
**Pull request by tpitts-CFI**
### Install instructions:

**Pre-Install checklist**
**If you haven't previously done this:**
1. `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash`
2. `nvm install 12.22.3`
3. `nvm alias default 12.22.3`
4. close workspace then open it again (this updates VScode with the new nvm default node/npm version)

**Install Checklist**

5. `ng update @angular/cli@10 @angular/core@10`
    * if ng update does not work:
    * 1. `npm uninstall -g @angular/cli`
    * 2. delete node_modules
    * 3. `npm cache verify`
    * 4. `npm install -g @angular/cli@10 @angular/core@10`
            * if this does not work, `npm install -g @angular/cli@10 @angular/core@10 --force`
6. `npm install`
7. `npm rebuild sass-loader`
8. if typescript version is NOT 3.9.10:
    * `npm uninstall --save typescript`
    * `npm install --save-dev typescript@3.9.10`
9. `ng update @angular/material@10`
    * if it says it can't:
    * 1. `ng update @angular/material@10 --force`

### **Warning**
* **RECOMMENDATION**: do not downgrade after this upgrade
* If you upgrade to this version and want to return to a previous version, follow these steps again using @7 for the ng update portions, downgrade to the previous node version using nvm alias default, and make sure you use typescript version 2.9.2.

### CODE REVIEW

File: kohese-portal/angular.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
             ],
             "scripts": [
               "./build/client/cache-worker/cache-bundle.js",
+              "node_modules/marked/lib/marked.js",
```
Comment (cfi-dphillips) [inline kohese-portal/angular.json:50]: Why is marked being added to the scripts?

File: kohese-portal/client/src/components/object-editor/format-object-editor/format-object-editor.component.spec.ts [hunk 2]
```diff
@@ -62,7 +63,9 @@ describe('FormatObjectEditorComponent', () => {
         FormsModule,
         BrowserAnimationsModule,
         MaterialModule,
-        MarkdownModule,
+        MarkdownModule.forRoot({
+          sanitize: SecurityContext.NONE
```
Comment (cfi-dphillips) [inline kohese-portal/client/src/components/object-editor/format-object-editor/format-object-editor.component.spec.ts:67]: Need to evaluate the implications of this.

File: kohese-portal/client/src/components/tree/default-tree/default-tree.component.spec.ts [hunk 3]
```diff
@@ -135,24 +137,21 @@ describe('Component: default-tree', () => {
     }
     expect(index).toEqual(-1);
 
-    TestBed.get(ActivatedRoute).params.next({ id: 'Item' });
-    tick();
-
     for (let j: number = 0; j < component.visibleRows.length; j++) {
-      if ('Item' === (component.visibleRows[j].object as ItemProxy).item.id) {
+      if ('Model-Definitions' === (component.visibleRows[j].object as ItemProxy).item.id) {
```
Comment (cfi-dphillips) [inline kohese-portal/client/src/components/tree/default-tree/default-tree.component.spec.ts:141]: What is the purpose of this change?

File: kohese-portal/angular.json (hunk 1 above)
Comment (tpitts-CFI) [reply kohese-portal/angular.json:50]: [ngx-markdown on npm](https://www.npmjs.com/package/ngx-markdown) has this as step 2 of its installation. The relevant excerpt is here:

As the library is using Marked parser you will need to add node_modules/marked/lib/marked.js to your application.
If you are using Angular CLI you can follow the angular.json example below...
```
"scripts": [
+ "node_modules/marked/lib/marked.js"
]
```

File: kohese-portal/client/src/components/object-editor/format-object-editor/format-object-editor.component.spec.ts (hunk 2 above)
Comment (tpitts-CFI) [reply kohese-portal/client/src/components/object-editor/format-object-editor/format-object-editor.component.spec.ts:67]: Setting the SecurityContext to none was necessary for our markdown to appear correctly. Many locations were not checked, but Document Row component's title's would not show correctly without this option set.
There is, however, another sanitization process that happens in one of the pipes that markdown goes through. This security context is also set to bypass (aka ignore security issues).

File: kohese-portal/client/src/components/tree/default-tree/default-tree.component.spec.ts (hunk 3 above)
Comment (tpitts-CFI) [reply kohese-portal/client/src/components/tree/default-tree/default-tree.component.spec.ts:141]: This has been resolved in angular-upgrade-12

document as:
Issue: default-tree component test synchronizes with selection does nothing (d122ec00-051d-11ec-9403-11f290226b38)

**Review by cfi-dphillips — APPROVED** [review]
